src/linter/yaml/UI5YamlLinter.ts

```diff
@@ -0,0 +1,87 @@
+import {LintMessageSeverity} from "../LinterContext.js";
+import LinterContext from "../LinterContext.js";
+import deprecatedLibraries from "../../utils/deprecatedLibs.js";
+import yaml from "js-yaml";
+import {DataWithPosition, fromYaml, getPosition} from "data-with-position";
+
+// file content schema of 'UI5Yaml' with only relevant properties
+interface UI5YamlContentSchema { // extend for further detections
+	framework: {
+		libraries: {
+			name: string;
+		}[];
+	};
+}
+
+interface UI5YamlContentSchemaWithPosInfo extends DataWithPosition {
+	framework?: {
+		libraries?: {
+			name: string;
+		}[];
+	};
+	positionKey?: {
+		end: {
+			column: number;
+			line: number;
+		};
+		start: {
+			column: number;
+			line: number;
+		};
+	};
+}
+
+export default class UI5YamlLinter {
+	#content = "";
+	#yamlContentWithPosInfo: UI5YamlContentSchemaWithPosInfo = {};
+	#resourcePath = "";
+	#context: LinterContext;
+
+	constructor(resourcePath: string, content: string, context: LinterContext) {
+		this.#content = content;
+		this.#resourcePath = resourcePath;
+		this.#context = context;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/require-await
+	async lint() {
+		try {
+			const source: UI5YamlContentSchema = this.#parseUI5Yaml(this.#content);
+			this.#analyzeUI5Yaml(source);
+		} catch (err) {
+			const message = err instanceof Error ? err.message : String(err);
+			this.#context.addLintingMessage(this.#resourcePath, {
+				severity: LintMessageSeverity.Error,
+				message,
+				ruleId: "ui5-linter-parsing-error",
+				fatal: true,
+			});
+		}
+	}
+
+	#parseUI5Yaml(fileContent: string): UI5YamlContentSchema {
+		// Create JS object from YAML content with position information
+		this.#yamlContentWithPosInfo = fromYaml(fileContent) as UI5YamlContentSchemaWithPosInfo;
+		// Convert YAML content to JS object
+		return yaml.load(fileContent) as UI5YamlContentSchema;
```

Comment:
I just debugged this and it looks like `this.#yamlContentWithPosInfo` already contains a JS-representation of the yaml content. Why is it necessary to parse the file again using `yaml.load`?

![Screenshot 2024-04-16 at 13 25 32](https://github.com/SAP/ui5-linter/assets/1589939/8c28f96c-97b8-4fb4-90ee-49a9f7f94bd5)

Reply:
e52efc0